src/backend/distributed/operations/shard_split.c

```diff
@@ -537,6 +508,10 @@ BlockingShardSplit(SplitOperation splitOperation,
 		InsertSplitChildrenShardMetadata(shardGroupSplitIntervalListList,
 										 workersForPlacementList);
 
+		/* create partitioning hierarchy, if any */
+		CreatePartitioningHierarchy(shardGroupSplitIntervalListList,
```

Comment:
is there a reason to do this after the data COPY?

Reply:
We can only do this after Metadata swap as to ATTACH to the new / correct parent (co-located) shard in 'GenerateAttachShardPartitionCommand'. There already will be an attached shard before that and we will fail due to overlapping partition.